wkcuber/cubing.py

```diff
@@ -156,8 +170,48 @@ def cubing_job(args):
                     ]
 
                 buffer = prepare_slices_for_wkw(slices, num_channels)
+                if downsampling_needed:
+                    logging.info(f"Downsampling buffer of size {buffer.shape}.")
+                    padding_size_for_downsampling = [
+                        (
+                            0,
+                            (
+                                required_multiple_of
+                                - (buffer_dimension_size % required_multiple_of)
+                            )
+                            % required_multiple_of,
```

Comment:
Why is this modulo operation necessary? `buffer_dimension_size % required_multiple_of` evaluates to a value with `value <= required_multiple_of`. That's why `required_multiple_of - (buffer_dimension_size % required_multiple_of)` has to be `< requried_multiple_of` which is why `% required_multiple_of` won't make any difference. Or am I missing something?

Reply:
`padding_size_for_downsampling` can be calculated more concisely by leveraging vectorization with numpy. Something along these lines:

```python
target_mag_np = np.array(target_mag.to_array())
buffer_shape_np = np.array(buffer.shape[1:])
padding_size_for_downsampling = target_mag_np - (buffer_shape_np % target_mag_np)
```

Reply:
Thanks for the numpy approach. It makes it way easier to read. :+1:  But I think that the second % is important. Imagine `buffer_shape=10` and `target_mag=2` then `buffer_shape % target_mag = 0` and therefore the `padding_size=2` even if no padding is needed. Therefore the second % is used to catch the case that there is no padding needed.

Reply:
Ah, now, I get it. When the first mod operation yields 0, the subtraction yields `padding_size` which has to be turned to zero. Good point :+1: However, I think, the overall operation can still be expressed more concise with: `-buffer_shape_np % required_multiple_of` :)